subsys/net/ip/tcp2.c

```diff
@@ -387,7 +387,9 @@ static bool tcp_options_check(void *buf, ssize_t len)
 
 		NET_DBG("opt: %hu, opt_len: %hu", (u16_t)opt, (u16_t)opt_len);
 
-		if (opt == TCPOPT_END || opt == TCPOPT_NOP) {
+		if (opt == TCPOPT_END) {
```

Comment:
@ rmstoi

We might also check if there's an extra options data past END, and if so, fail options check. In principle, we have all the data for this here. Also, for your own judgement.

Reply:
> We might also check if there's an extra options data past END, and if so, fail options check. In principle, we have all the data for this here. Also, for your own judgement.

Thanks for the improvement ideas. If you don't mind let's keep those in the backlog for the future. Lets implement these once we have more spare time. For this iteration the solution is similar to TCP1 and will work for most of the cases.

Reply:
@rmstoi 

Thanks and sure. I don't mind.